examples/general/abs_home_path/src/lib.rs

```diff
@@ -178,52 +124,94 @@ fn ui() {
         .run();
 }
 
+// Combined test for context allowances (build script and test context)
 #[test]
-fn build_script_allowance() {
+fn context_allowance() {
     use dylint_internal::CommandExt;
     use std::{
         io::{Write, stderr},
         path::Path,
         process::Command,
+        process::Output,
     };
 
-    // Skip test if repository is not stored in the user's home directory
+    struct TestCase {
+        manifest_path: &'static str,
+        context_name: &'static str,
+        assert_fn: fn(&Output),
+    }
+
+    let test_cases = [
+        TestCase {
+            manifest_path: "ui_build_script/Cargo.toml",
+            context_name: "build script",
+            assert_fn: |output: &Output| {
+                let stderr = String::from_utf8_lossy(&output.stderr);
+                assert!(
+                    output.status.success(),
+                    "Cargo dylint check failed in build script: {stderr}"
+                );
+                assert!(
+                    !stderr.contains("this path might not exist in production"),
+                    "The abs_home_path lint was incorrectly triggered in a build script: {stderr}"
+                );
+            },
+        },
+        TestCase {
+            manifest_path: "ui_test/Cargo.toml",
+            context_name: "test context",
+            assert_fn: |output: &Output| {
+                let stderr = String::from_utf8_lossy(&output.stderr);
+                // We expect warnings for non-test functions but not for test functions
+                assert!(
+                    stderr.contains("this path might not exist in production"),
+                    "Expected the abs_home_path lint to trigger for non-test functions in test context: {stderr}"
+                );
+                // Verify the test detected non-test functions correctly
+                assert!(
+                    stderr.contains("non_test_function_with_home_path")
+                        || stderr.contains("src/main.rs:9:13"),
+                    "Expected warning in non-test function in test context: {stderr}"
+                );
+                // Verify that test functions weren't warned about
+                assert!(
+                    !stderr.contains("test_function_with_home_path")
+                        && !stderr.contains("src/main.rs:15:13"),
+                    "The abs_home_path lint should not trigger in test functions: {stderr}"
+                );
```

Comment:
This feels fragile to me. If the test name changes, or lines are added to or removed from the file, the check will succeed.

I think I would prefer a check something like this:
```rs
stderr.lines().all(|line| !line.contains("src/main.rs") || line.contains("src/main.rs:9:13"))
```
In other words, the only line that is flagged is the one you already checked for.